### PR TITLE
#85 directors history

### LINKED ISSUE #85: Client: Implement Board of directors history
Implementation of a list showing the current board of directors and the history of past board of directors members. In addition, a filter, sorting and search is implemented.

### PR DESCRIPTION
**Pull request by lisa-bai**
created directors history

closes #85

### CODE REVIEW

File: client/src/ts/members/DirectorHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+      headers: {Authorization : `Bearer ${localStorage.getItem("token")}`}
+    })
+    .then((res) => {
+      if (res.status === 200){
+        if(mounted) {
+          setdirectors(res.data);
+        }
+      }
+    }).catch((error) => {
+      console.log(error);
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:176]: no console logs

File: client/src/ts/members/DirectorHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+      filtereddirectors = filtereddirectors.filter(director => {
+        return director.kuerzel === kuerzelFilter;
+      });
+    }
+
+    // Filters by search input
+    filtereddirectors = filtereddirectors.filter(director => {
+      return (
+        director.vorname.toLowerCase().includes(searchFilter.toLowerCase()) ||
+        director.nachname.toLowerCase().includes(searchFilter.toLowerCase())
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:235]: trailing whitespace

File: client/src/ts/members/DirectorHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+        case "down" : {
+          return <ExpandMore/>;
+        }
+      }
+  };
+
+  // The additional filters
+  const additionalFilters = (
+    <div>
+
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:302]: trailing whitespace

File: client/src/ts/members/DirectorHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+          <MenuItem value={"NET"}>NET</MenuItem>
+          <MenuItem value={"QM"}>QM</MenuItem>
+          <MenuItem value={"F&R"}>F&R</MenuItem>
+          <MenuItem value={"FK"}>FK</MenuItem>
+          <MenuItem value={"MIT"}>MIT</MenuItem>
+          <MenuItem value={"MAR"}>MAR</MenuItem>
+          <MenuItem value={"IT"}>IT</MenuItem>
+        </TextField>
+      </Grid>
+
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:322]: trailing whitespace

File: client/src/ts/members/DirectorHistory.tsx [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+              <MenuItem value={"RL F&R"}>F&R</MenuItem>
+              <MenuItem value={"RL FK"}>FK</MenuItem>
+              <MenuItem value={"RL MIT"}>MIT</MenuItem>
+              <MenuItem value={"RL MAR"}>MAR</MenuItem>
+              <MenuItem value={"RL IT"}>IT</MenuItem>
+              <MenuItem value={"1V"}>1V</MenuItem>
+            </TextField>
+          </Grid>
+        </Grid>
+
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:362]: trailing whitespace

File: client/src/ts/members/DirectorHistory.tsx [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+            <TableBody>
+              {getFilteredandSortedDirectors().map((director, index) => (
+                <TableRow hover key={index}>
+                  <TableCell
+                    component="th"
+                    scope="row"
+                  >
+                    {`${director.vorname} ${director.nachname}`}
+                  </TableCell>
+
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:410]: trailing whitespace

File: client/src/ts/members/DirectorHistory.tsx [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {getFilteredandSortedDirectors().map((director, index) => (
+                <TableRow hover key={index}>
+                  <TableCell
+                    component="th"
+                    scope="row"
+                  >
+                    {`${director.vorname} ${director.nachname}`}
```
Comment (johannesfbraun) [inline client/src/ts/members/DirectorHistory.tsx:408]: I think a link to the user profile, when clicking on the name would be good

**Review by johannesfbraun — CHANGES_REQUESTED** [review]
Well done only minor changes :)

File: client/src/ts/members/DirectorHistory.tsx (hunk 1 above)
Comment (LennartRukower) [reply client/src/ts/members/DirectorHistory.tsx:176]: Until now console logs were the standard way of dealing with errors.

But after the implementation of Snackbars (see #78) the usage of Snackbars for displaying errors is the way to go.

File: client/src/ts/members/DirectorHistory.tsx [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+const DirectorHistory: React.FunctionComponent = () => {
+  const classes = useStyles();
+
+  const [additionalFiltersState, setAddtionalFiltersState] = useState(false);
+  const [directors, setdirectors] = useState<Director[]>([]);
+  const [searchFilter, setSearchFilter] = useState<string>("");
+  const [statusFilter, setStatusFilter] = useState<string>("");
+  const [kuerzelFilter, setkuerzelFilter] = useState<string>("");
+  const [sortOption, setSortOption] = useState<string>("");
+
```
Comment (LennartRukower) [inline client/src/ts/members/DirectorHistory.tsx:159]: Why is here an empty line?

File: client/src/ts/members/DirectorHistory.tsx (hunk 3 above)
Comment (LennartRukower) [reply client/src/ts/members/DirectorHistory.tsx:302]: No empty line

File: client/src/ts/members/DirectorHistory.tsx (hunk 4 above)
Comment (LennartRukower) [reply client/src/ts/members/DirectorHistory.tsx:322]: No empty line

File: client/src/ts/members/DirectorHistory.tsx (hunk 5 above)
Comment (LennartRukower) [reply client/src/ts/members/DirectorHistory.tsx:362]: No empty line

File: client/src/ts/members/DirectorHistory.tsx (hunk 6 above)
Comment (LennartRukower) [reply client/src/ts/members/DirectorHistory.tsx:410]: No empty line

**Review by LennartRukower — CHANGES_REQUESTED** [review]
Very good first pr!
Please fix the small formatting mistakes and rerequest my review after that.

Please check if your typescript linter (TSLint) is activated in VSC. With this extension the formatting mistakes are visually indicated by a warning.

File: client/src/ts/members/DirectorHistory.tsx (hunk 2 above)
Comment (johannesfbraun) [reply client/src/ts/members/DirectorHistory.tsx:216]: trailing whitespace (no-trailing-whitespace)

Comment (johannesfbraun) [reply client/src/ts/members/DirectorHistory.tsx:216]: trailing whitespace (no-trailing-whitespace)

**Review by johannesfbraun — CHANGES_REQUESTED** [review]
looks good; only the trailing whitespace is still there

**Review by jbt-it — APPROVED** [review]
LGTM

**Pull request merged**